scripts/clean-data.py

```diff
@@ -132,11 +166,28 @@ def clean_fqn_test():
     'fastparse.Api.P(scala.Functi_on0,sourcecode.Name)fastparse.core.Parser'
     '''
 
+clean_signature = compose(
+    remove_arrow,
+    remove_hashtags,
+    remove_L_notation,
+    remove_trailing_dot,
+    remove_leading_root
+)
+def clean_signature_test():
+    '''
+    An empty function to trigger doctest.
+    clean_signature agglomerates various cleaning functions to clean the resolved type name from semanticdb.
+
+    >>> clean_signature('[T, R] => (x: _root_.scala.Tuple2#[_root_.scala.concurrent.Future#[T], _root_.scala.concurrent.Future#[R]])(implicit executionContext: _root_.scala.concurrent.ExecutionContext#): _empty_.PipeableFuture#[T, R]')
+    '[T, R](x: scala.Tuple2[scala.concurrent.Future[T], scala.concurrent.Future[R]])(implicit executionContext: scala.concurrent.ExecutionContext): PipeableFuture[T, R]'
+    '''
+
 def clean_param_row(row):
     row["fqn"] = clean_fqn(row["fqn"])
     row["fqfn"] = extract_function_name(row["fqn"])
+    row["name"] = extract_plain_name(row["fqn"])
     row["fqparamlist"] = extract_parameter_list(row["fqn"])
```

Comment:
Is this actually useful? I think the full signature is probably enough. For anything more elaborate we will need to break this down to function arguments in a new table

Reply:
I honestly don't know, it depends on what the usage looks like on the R side, we haven't talked about that yet.

Reply:
i think this can be easily parsed in R